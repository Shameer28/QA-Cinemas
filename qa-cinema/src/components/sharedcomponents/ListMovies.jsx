
import PosterLink from "./PosterLink";

import movieUtils from "./../../utils/movies";
import Container from 'react-bootstrap/Container'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import { useState, useEffect } from "react";

const ListMovies = ({ sort }) => {

	const [data, setData] = useState([]);
	const [err, setError] = useState("");
	const [loading, setLoading] = useState(true);

	useEffect(() => {
		movieUtils.getAll().then((resp) => {
			let info = resp.data;

			if (sort) {
				// sort the info here
			}

			setLoading(false);
			setData(info);
		}).catch((err) => {
			setLoading(false);
			setError(err);
		})

		// eslint-disable-next-line react-hooks/exhaustive-deps
	}, []);

	if (loading) {
		return <p>Loading</p>
	}

	if (err !== "") {
		return (<p>
			Oh no looks like something went wrong!
		</p>)
	}


	return (
		<div>
			<Row md={4}>
				{data.map(x =>
					<div>
						<Col xs={4}>
							<div className="image">
								<PosterLink image={x.image} _id={x._id} title={x.title}/>
								<br></br>
							</div>							
							<br></br>
						</Col>
					</div>

				)}
			</Row>
			<br></br>
		</div>
	);

}

export default ListMovies;