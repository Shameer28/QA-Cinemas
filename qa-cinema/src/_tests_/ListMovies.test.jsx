import ListMovies from "../components/sharedcomponents/ListMovies";
import { create } from 'react-test-renderer';


describe(`List Movies Page testing`, () => {

    it(`Should match the snapshot`, () => {
        const instance = create(<ListMovies />).toJSON();
        expect(instance).toMatchSnapshot();
    })
})