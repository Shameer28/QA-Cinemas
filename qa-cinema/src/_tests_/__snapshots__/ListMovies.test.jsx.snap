// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Movies Page testing Should match the snapshot 1`] = `
<p>
  Loading
</p>
`;
